### PR TITLE
Automated approver tests

### PR DESCRIPTION
**Pull request by dekiel**
**Description**

Changes proposed in this pull request:

- Added tests for automated approver tool.

**Related issue(s)**
See #7285

### CODE REVIEW

File: cmd/external-plugins/automated-approver/plugin_test.go [hunk 1]
```diff
@@ -0,0 +1,389 @@
...
+		handler                 main.HandlerBackend
+		ghc                     *fakegithub.FakeClient
+		eventPayloadFilePath    string
+		oldEventPayloadFilePath string
+		eventHandler            func(*externalplugin.Plugin, externalplugin.Event)
+		wg                      sync.WaitGroup
+	)
+	BeforeEach(func() {
+		logger, level := consolelog.NewLoggerWithLevel()
+		// level.SetLevel(zapcore.DebugLevel)
```
Comment (akiioto) [inline cmd/external-plugins/automated-approver/plugin_test.go:52]: If not used, then maybe remove?

Comment (dekiel) [reply cmd/external-plugins/automated-approver/plugin_test.go:52]: I've left it there to easily enable debug logging when we need it for troubleshooting. There is no flag to do it.

**Review by akiioto — APPROVED** [review]

**Pull request merged**